doc/getting-started/gsg_quick_setup.sh

```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+# Copyright (C) 2019 Intel Corporation.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# This script provides a quick and automatic setup of the SOS or UOS.
+# You should run this script with root privilege since it will modify various system parameters.
+#
+# Usages:
+#   Upgrade SOS to 28100 without reboot, it's highly recommended so that you can check configurations after upgrade SOS.
+#     sudo <script> -s 28100 -d
+#   Upgrade SOS to 28100 with specified proxy server
+#     sudo <script> -s 28100 -p <your proxy server>:<port>
+#   Upgrade UOS to 28100 with specified proxy server
+#     sudo <script> -u 28100 -p <your proxy server>:<port>
+#   Upgrade UOS to 28100 without downloading UOS image, you should put UOS image in /root directory previously.
+#     sudo <script> -u 28100 -k
+
+function print_help()
+{
+    echo "Usage:"
+    echo "Launch this script as: sudo $0 -s 28100"
+    echo -e "\t-s to upgrade SOS"
+    echo -e "\t-u to upgrade UOS"
+    echo -e "\t-p to specify a proxy server"
+    echo -e "\t-m to use swupd mirror url"
+    echo -e "\t-k to skip downloading UOS; if enabled, you have to download UOS img firstly before upgrading, default is off"
+    echo -e "\t-d to disable reboot device so that you can check the configurations after upgrading SOS"
+    exit 1
+}
+
+# get clear linux version previously
+source /etc/os-release
+# switcher for download UOS function
+skip_download_uos=0
+# switcher for disabling reboot device
+disable_reboot=0
+
+# acrn.conf path
+acrn_conf_path=/usr/share/acrn/samples/nuc/acrn.conf
+# acrn.efi path
+acrn_efi_path=/usr/lib/acrn/acrn.efi
+
+function upgrade_sos()
+{
+    # Check SOS version
+    [[ `echo $sos_ver | awk '{print length($0)}'` -ne 5 ]] && echo "Please input right SOS version to upgrade" && exit 1
+    [[ $VERSION_ID -gt $sos_ver ]] && echo "You're trying to install an older version of Clear Linux." && exit 1
+
+    echo "Upgrading SOS..."
+
+    # setup mirror and proxy url while specified with m and p options
+    [[ -n $mirror ]] && echo "Setting swupd mirror to: $mirror" && swupd mirror -s $mirror
+    [[ -n $proxy ]] && echo "Setting proxy to: $proxy" && export https_proxy=$proxy
+
+    # You should set EFI path before upgrading SOS.
+    efi_partition=/dev/sda1
+    # Check EFI path exists.
+    [[ ! -b $efi_partition ]] && echo "Please choose the right EFI path firstly." && exit 1
```

Comment:
How can the user set the ESP to the right value? Is editing the script the only option?

We should add a note explaining this

Reply:
Yes, so far it's the only option to edit script to set ESP.
I have also added a note at the head of the script to explain user should set the right ESP before running script.

Reply:
We should really make this a command-line parameter and the explanation should be included in the "usage" section. People do not typically read scripts before running them. It's great to have details and explanations in them but for something that really has to be provided by the user, a parameter should be used (as opposed to editing the script).

Reply:
Thank you very much for suggestion, I have added a new parameter "-e" for user to specify the right ESP and also added "note" into print_help function.